lib/Interpreter/Paths.cpp

```diff
@@ -106,53 +106,38 @@ namespace platform {
     return std::string(Buffer.str());
   }
 
-#if defined(LLVM_ON_UNIX)
-  static void DLErr(std::string* Err) {
-    if (!Err)
-      return;
-    if (const char* DyLibError = ::dlerror())
-      *Err = DyLibError;
+#if defined(_WIN32)
+  void* DLOpen(const std::string& Path, std::string& Err) {
+    auto lib = llvm::sys::DynamicLibrary::getLibrary(Path.c_str(), &Err);
+    return lib.getOSSpecificHandle();
   }
 
-  void* DLOpen(const std::string& Path, std::string* Err /* = nullptr */) {
-    void* Lib = ::dlopen(Path.c_str(), RTLD_LAZY | RTLD_GLOBAL);
-    DLErr(Err);
-    return Lib;
+  void DLClose(void* Lib, std::string& Err) {
+    auto dl = llvm::sys::DynamicLibrary(Lib);
+    llvm::sys::DynamicLibrary::closeLibrary(dl);
+    if (Err.empty()) {
+      Err = std::string();
+    }
+  }
+#elif defined(LLVM_ON_UNIX)
+  static void DLErr(std::string& Err) {
```

Comment:
Why not returning a string here?

Reply:
I did not change that but we can if that is necessary.

Reply:
ok, let's keep it as is.